documentation/monitoring.md

```diff
@@ -0,0 +1,110 @@
+# Monitoring
+
+hwbench is using a specific monitoring engine to collect data from several sources :
+- BMC
+- PDU
+- Turbostat
+
+Regarding the source, power, frequencies, thermal or cooling metrics are collected.
+
+# Concept
+During each benchmark, and if the monitoring is enabled, metrics are collected every 2 seconds and aggregated every 10 seconds to get statistics over this period of time.
+
+At the end of the benchmark the monitoring metrics are added in the a result file. hwgraph will use them to plot how these components behave during the benchmark.
+
+# Usage
+To enable the monitoring feature, just set the `monitor` directive in the configuration file.
+
+As per this release, only `monitor=all` is supported. In future releases, it will be possible to list the source to be considered.
+
+# Configuration file
+When monitoring is enabled, the `-m <config_file>` option must be used to describe the server's configuration.
+This file is separated from the job file as it could be specific to each host.
+
+Each source is defined like the following :
+
+```
+  [section_name]
+  username=<username>
+  password=<password>
+  type=<type>
+ ```
+
+ ## BMC
+ When defining a BMC, the `type` must be set to `BMC`. The `section_name` could the hardware vendor name like `DELL`, `HPE`, or `default`.
+
+ A typical example looks like :
+ 
+ ```
+ #[HPE]
+username=Administrator
+password=YOURPASSWORD
+type=BMC
+ ```
+
+**Note**: there is no need to define the URL of the BMC, it will be automatically detected at runtime.
```

Comment:
Maybe say how it's supposed to be detected (through local IPMI?)

Also, in the future it might be useful to add a URL for servers that don't have a BMC<->Host local IPMI link.

Reply:
I added a note to clarify this